Api doc hub 1.0 announcement (#12)
spacetelescope/mast-blog: _posts/2025-04-29-api-docs-hub-1.0-post.md

---
layout: post
read_time: true
show_date: true
title: "MAST API Documentation Hub 1.0"
date: 2025-04-29
img: posts/19000101/template_image.png
tags: [tutorials, release]
category: Documentation
author: MAST Staff
description: "Search API documentation across MAST web services"
---

MAST has just launched a centralized searchable web portal indexing collected documentation for our APIs. You can browse through summaries of supported services or enter search keywords (for example: mission names, interface standards like "cone search", or service features like "cutouts") to retrieve a list of related services with direct links into their docs. 

> This 1.0 release focuses on the many OpenAPI standard web services MAST hosts on our Python platform, providing quick links to project homepages and to Sphinx API documentation in a familiar Read the Docs style, with included SwaggerUI pages for interacting with each service live. We are excited to expand the API Documentation Hub knowledge base to other service platforms and documentation repositories supported by MAST in future versions. Searchable contents will be automatically updated along with every included service release.

Reviewer: ```suggestion
This 1.0 release focuses on the many OpenAPI standard web services MAST hosts, providing quick links to project homepages and to API documentation. Searchable contents are updated along with every service release. We are looking forward to expanding this API Documentation Hub to include other service platforms and documentation repositories supported by MAST.
```

Omit implementation details

Try to condense it somewhat

I'm a little on the fence about including the sentence about the future work here.

Author: The choice to include the implementation details was intended to highlight the tech behind it for users who may know what those tools are but hadn't looked at our docs in a while. It could be a bit much. @ttdu, what level do you expect for these posts? I saw a tag in the repo for deep dives on tech stacks, and I know this isn't it.

I am also on the fence about promising future work. I do want it to answer 'why isn't $blah in here' with intentions but not undue specific commitments.

Reviewer 2: "We are excited to expand the API Documentation Hub knowledge base to other service platforms and documentation repositories supported by MAST in future versions." --> "Additional MAST services, notebooks, and/or documentation may be included in the future, as resources allow."
